### PR TITLE
API fully functional and integrated with front end

### PR DESCRIPTION
**Pull request by sunikkim**
- API now supports all front-end functions including POST, PUT, etc. Tested with the app.
- Got code coverage to 97% by including erroring/broken routes in my code
- Implemented K6 load testing - getting an average http_req_duration of 4.81ms for my GET request, which seems good so far. 
- My API will now send evergreen data if a request fails, or an invalid productId is used. So no request can break the app (so far!)
- Implemented Sort By functionality
- Implemented count functionality (if user supplies a '?count=' query, my API will limit the returned reviews to that count)

NEXT STEPS:
- optimizing, doing more load/stress testing, thinking about how to reduce the amount of duplicate data in my DB query and if I can write more efficient helper functions. then on to deployment. considering trying a mongo implementation if I have time as well.

### CODE REVIEW

File: server/reviewsHelpers.js [hunk 1]
```diff
@@ -139,6 +140,14 @@ const formatReviews = (result, productId) => {
     }
   }
 
+  if (sortMethod === 'newest') {
```
Comment (rlawrence9) [inline server/reviewsHelpers.js:143]: dig the clean use of sort here 🤙

File: test/k6Tests.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  sleep(1);
+
+  const checkRes = check(res, {
+    'status is 204 (review marked helpful)': (r) => r.status === 204,
+    'status is 400 (could not mark review helpful)': (r) => r.status === 400,
+  });
+};
+
+export const options = {
```
Comment (rlawrence9) [inline test/k6Tests.js:92]: wonder if there is a way to have custom options per test?  i know your tests dont particularly need it, but it would be cool to be able to run like a couple different tests.  maybe one that tests what happens when your service goes from like 1 person to 1k people in like 1 second and also just the regular load test that tests the consistency of RPS

File: test/k6Tests.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  scenarios: {
+    constant_request_rate: {
+      executor: 'constant-arrival-rate',
+      rate: 1000,
```
Comment (rlawrence9) [inline test/k6Tests.js:96]: are you able to hit this rate with 225 vus?  i noticed on my testing when i put the rate higher then the amount of VUs could handle, it would through errors in console of k6 test.  For me 225 Vus could handle a maximum of like 218 RPS

File: test/reviewsTest.js [hunk 4]
```diff
@@ -81,31 +139,31 @@ describe('GET /reviews', () => {
...
     expect(response.body.product).to.eql('1');
     expect(response.body.results.length).to.not.eql(0);
   });
 
   it('returns correct reviews in correct format for valid product id 100000', async () => {
-    const response = await request.get('/product/100000');
+    const response = await request.get('/product/100000?sort=helpfulness');
     expect(response.body.product).to.eql('100000');
     expect(response.body.results.length).to.not.eql(0);
   });
```
Comment (rlawrence9) [inline test/reviewsTest.js:169]: clean looking tests! they are definitely testing your functionality to the T!!! nice work!

**Review by rlawrence9 — APPROVED** [review]
overall really nice work sunik!
